packages/flutter/lib/src/material/text_field.dart

```diff
@@ -782,7 +782,7 @@ class _TextFieldState extends State<TextField> implements TextSelectionGestureDe
     if (widget.maxLength > 0) {
       // Show the maxLength in the counter
       counterText += '/${widget.maxLength}';
-      final int remaining = (widget.maxLength - currentLength).clamp(0, widget.maxLength);
+      final int remaining = (widget.maxLength - currentLength).clamp(0, widget.maxLength) as int;
```

Comment:
Similar to the `.clamp(...) as double`, I think here it'd be safer to say `.clamp(...).toInt()`.

Reply:
same comment as above for `as double` vs. `.toDouble()`